src/main/java/edu/itu/cavabunga/business/CalendarManagerServiceImpl.java

```diff
@@ -0,0 +1,673 @@
+package edu.itu.cavabunga.business;
+
+import edu.itu.cavabunga.core.entity.Component;
+import edu.itu.cavabunga.core.entity.Parameter;
+import edu.itu.cavabunga.core.entity.Participant;
+import edu.itu.cavabunga.core.entity.Property;
+import edu.itu.cavabunga.core.entity.component.ComponentType;
+import edu.itu.cavabunga.core.entity.parameter.ParameterType;
+import edu.itu.cavabunga.core.entity.participant.ParticipantType;
+import edu.itu.cavabunga.core.entity.property.PropertyType;
+import edu.itu.cavabunga.core.services.IcalService;
+import edu.itu.cavabunga.core.services.ParticipantService;
+import edu.itu.cavabunga.exception.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class CalendarManagerServiceImpl implements CalendarManagerService {
+    @Autowired
+    private IcalService icalService;
+
+    @Autowired
+    private ParticipantService participantService;
+
+    @Override
+    public void addParticipant(Participant participant){
+         if(participant.getUserName().isEmpty() || participant.getUserName() == null){
+             throw new ParticipantConflict(1000,"Participant username cannot be empty");
```

Comment:
Do we really need error code? If yes, tracking the error codes hard in this case. Maybe we can consider removing.

Reply:
For instance, in many place "checking participant exists" operation happens. Do determine where exactly exception comes from is important. <imho>

Reply:
You can use stack trace to determine where exactly exception comes from. Giving a random number does not help in my opinion.